packages/block-library/src/comment-template/index.php

```diff
@@ -25,13 +25,14 @@ function block_core_comment_template_render_comments( $comments, $block ) {
 
 	$content = '';
 	foreach ( $comments as $comment ) {
-
-		$block_content = ( new WP_Block(
-			$block->parsed_block,
-			array(
-				'commentId' => $comment->comment_ID,
-			)
-		) )->render( array( 'dynamic' => false ) );
+		$comment_id           = $comment->comment_ID;
+		$filter_block_context = static function( $context ) use ( $comment_id ) {
+			$context['commentId'] = $comment_id;
```

Comment:
Have you tried using `array_merge` with the `$block->context` instead?

Reply:
Ah, I haven't. You mean inside of the filter? I _think_ that the context that this block receives should be already present in the `$context` that's available to the `render_block_context` filter, no? 🤔

Reply:
I meant with the old approach that wrongly assumes that the block context can be fully overwritten.